cpp/bench/matrix/select.cu

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../../test/matrix/select.cuh"
```

Comment:
I'd prefer not to mix/include test files in benchmark code (and vice versa)

Reply:
There's a great deal of shared code in this file! Would you consider adding some shared test+bench header-only component? As far as I know, at the moment we have only one more place where we explicitly refer to a test header from the benchmarks ([bench/neighbors/refine.cu#L39](https://github.com/rapidsai/raft/blob/aec0948305d708323a4781cc16c5974b4ecf1a3c/cpp/bench/neighbors/refine.cu#L39)), but there could definitely be more.